File: src/main/java/shopping/domain/member/Member.java
package shopping.domain.member;

import jakarta.persistence.*;

@Entity
public class Member {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String email;

    @Column(unique = false)
    private String password;

    @Enumerated(EnumType.STRING)
    private Role role;

    protected Member() {}

    public Member(String email, String password) {
        this(email, password, Role.CONSUMER);
    }

    public Member(String email, String password, Role role) {
        this.email = email;
        this.password = password;
        this.role = role;
    }

    public Long getId() {
        return id;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }

    public Role getRole() {
        return role;
    }

    public static Member create(String email, String password) {
        return new Member(email, password, Role.CONSUMER);
    }
}

File: src/main/kotlin/shopping/domain/member/Member.kt
package shopping.domain.member

import jakarta.persistence.Entity
import jakarta.persistence.Table
import org.hibernate.annotations.SQLDelete
import shopping.support.config.jpa.BaseEntity
import shopping.support.error.CoreException
import shopping.support.error.ErrorType

@Table(name = "members")
@Entity
@SQLDelete(sql = "UPDATE members SET deleted_at = NOW() WHERE id = ?")
class Member(
    email: String,
    password: String
) : BaseEntity() {
    var email: String = email
        protected set

    var password: String = password
        protected set

    init {
        if (email.isBlank()) throw CoreException(ErrorType.BAD_REQUEST, "이메일은 비어있을 수 없습니다.")
        if (password.isBlank()) throw CoreException(ErrorType.BAD_REQUEST, "비밀번호는 비어있을 수 없습니다.")
    }

    fun checkPassword(password: String): Boolean {
        return password.equals(this.password)
    }

}